File: app/src/main/java/com/cyl/musiclake/ui/music/online/adapter/TopListAdapter.java
package com.cyl.musiclake.ui.music.online.adapter;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.cyl.musiclake.R;
import com.cyl.musiclake.api.netease.NeteaseList;
import com.cyl.musiclake.api.netease.NeteaseMusic;
import com.cyl.musiclake.bean.Music;
import com.cyl.musiclake.bean.Playlist;
import com.cyl.musiclake.utils.CoverLoader;

import java.util.List;

/**
 * 作者：yonglong on 2016/8/10 21:36
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
public class TopListAdapter extends BaseQuickAdapter<Playlist, BaseViewHolder> {

    public TopListAdapter(List<Playlist> list) {
        super(R.layout.item_top_list, list);
    }

    @Override
    protected void convert(BaseViewHolder helper, Playlist neteaseList) {
        if (neteaseList.getName() == null)
            return;
        helper.setText(R.id.title, neteaseList.getName());
        if (neteaseList.getCoverUrl() == null)
            return;
        CoverLoader.loadImageView(mContext, neteaseList.getCoverUrl(), helper.getView(R.id.iv_cover));
        List<Music> musicLists = neteaseList.getMusicList().subList(0, 3);
        if (musicLists.size() >= 1 && musicLists.get(0).getTitle() != null
                && musicLists.get(0).getArtist() != null) {
            helper.setText(R.id.tv_music_1, (mContext.getString(R.string.song_list_item_title_1,
                    musicLists.get(0).getTitle(), musicLists.get(0).getAlbum())));
        } else {
            helper.setText(R.id.tv_music_1, "");
        }
        if (musicLists.size() >= 2 && musicLists.get(1).getTitle() != null
                && musicLists.get(1).getArtist() != null) {
            helper.setText(R.id.tv_music_2, mContext.getString(R.string.song_list_item_title_2,
                    musicLists.get(1).getTitle(), musicLists.get(1).getAlbum()));
        } else {
            helper.setText(R.id.tv_music_2, "");
        }
        if (musicLists.size() >= 3 && musicLists.get(2).getTitle() != null
                && musicLists.get(2).getArtist() != null) {
            helper.setText(R.id.tv_music_3, mContext.getString(R.string.song_list_item_title_3,
                    musicLists.get(2).getTitle(), musicLists.get(2).getAlbum()));
        } else {
            helper.setText(R.id.tv_music_3, "");
        }
    }

    private String getAuthor(List<NeteaseMusic.ArtistsBeanX> artists) {
        if (artists.size() == 0) return null;

        StringBuilder artist = new StringBuilder(artists.get(0).getName());
        for (int i = 1; i < artists.size(); i++) {
            artist.append(",").append(artists.get(i).getName());
        }
        return artist.toString();
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/discover/TopListAdapter.kt
package com.cyl.musiclake.ui.music.discover

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.cyl.musicapi.netease.BannerBean
import com.cyl.musiclake.MusicApp
import com.cyl.musiclake.R
import com.cyl.musiclake.api.music.MusicApi
import com.cyl.musiclake.api.music.MusicUtils
import com.cyl.musiclake.api.music.MusicUtils.PIC_SIZE_NORMAL
import com.cyl.musiclake.bean.Album
import com.cyl.musiclake.bean.Playlist
import com.cyl.musiclake.common.Constants
import com.cyl.musiclake.common.Extras
import com.cyl.musiclake.common.NavigationHelper
import com.cyl.musiclake.player.PlayManager
import com.cyl.musiclake.ui.music.mv.MvDetailActivity
import com.cyl.musiclake.utils.CoverLoader
import com.cyl.musiclake.utils.Tools
import com.zhouwei.mzbanner.holder.MZViewHolder
import org.jetbrains.anko.startActivity

/**
 * 作者：yonglong on 2016/8/10 21:36
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
class TopListAdapter(list: List<Playlist>) : BaseQuickAdapter<Playlist, BaseViewHolder>(R.layout.item_top_list, list) {

    override fun convert(helper: BaseViewHolder, neteaseList: Playlist) {
        if (neteaseList.name == null)
            return
        helper.setText(R.id.title, neteaseList.name)
        if (neteaseList.coverUrl == null)
            return
        val url = when (neteaseList.type) {
            Constants.PLAYLIST_WY_ID -> MusicUtils.getAlbumPic(neteaseList.coverUrl, Constants.NETEASE, PIC_SIZE_NORMAL)
            else -> neteaseList.coverUrl
        }
        CoverLoader.loadImageView(mContext, url, helper.getView(R.id.iv_cover))
    }
}

class TopPlaylistAdapter(list: List<Playlist>) : BaseQuickAdapter<Playlist, BaseViewHolder>(R.layout.item_play_list, list) {

    override fun convert(helper: BaseViewHolder, playlist: Playlist) {
        val url = when (playlist.type) {
            Constants.PLAYLIST_WY_ID -> MusicUtils.getAlbumPic(playlist.coverUrl, Constants.NETEASE, PIC_SIZE_NORMAL)
            else -> playlist.coverUrl
        }
        CoverLoader.loadImageView(mContext, url, helper.getView(R.id.iv_cover))
        helper.setText(R.id.tv_title, playlist.name)
        helper.setText(R.id.tv_playCount, "播放次数：${playlist.playCount}")
    }
}

/**
 * 轮播图viewHolder
 */
class BannerViewHolder(val activity: Activity) : MZViewHolder<BannerBean> {
    private var mImageView: ImageView? = null
    private var mTypeTitle: TextView? = null
    override fun createView(context: Context): View {
        // 返回页面布局
        val view = LayoutInflater.from(context).inflate(R.layout.item_banner, null)
        mImageView = view.findViewById(R.id.banner_image) as ImageView
        mTypeTitle = view.findViewById(R.id.banner_title_tv) as TextView
        return view
    }

    override fun onBind(context: Context, position: Int, data: BannerBean?) {
        // 数据绑定
        CoverLoader.loadImageView(context, data?.picUrl, mImageView)
        mTypeTitle?.text = data?.typeTitle
        mImageView?.setOnClickListener {
            when {
                data?.targetType == "3000" -> Tools.openBrowser(MusicApp.getAppContext(), data.url)
                data?.targetType == "10" -> {
                    //专辑
                    NavigationHelper.navigateToPlaylist(activity, Album().apply {
                        albumId = data.targetId
                        type = Constants.NETEASE
                    }, null)
                }
                data?.targetType == "1000" -> {
                    //歌单
                    NavigationHelper.navigateToPlaylist(activity, Playlist().apply {
                        pid = data.targetId
                        type = Constants.PLAYLIST_WY_ID
                    }, null)
                }
                data?.targetType == "1004" -> {
                    //mv
                    context.startActivity<MvDetailActivity>(Extras.MV_ID to data.targetId)
                }
                data?.targetType == "1" -> {
                    //单曲
                    MusicApi.loadSongDetailInfo(Constants.NETEASE, data.targetId) { result ->
                        PlayManager.playOnline(result)
                    }
                }
            }
        }
    }
}


class PlaylistAdapter(playlists: List<Playlist>) : BaseQuickAdapter<Playlist, BaseViewHolder>(R.layout.item_playlist, playlists) {

    override fun convert(helper: BaseViewHolder, playlist: Playlist) {
        helper.setText(R.id.tv_name, playlist.name)
        CoverLoader.loadImageView(mContext, playlist.coverUrl, helper.getView(R.id.iv_cover))
        helper.setText(R.id.tv_num, playlist.musicList.size.toString() + "首")
    }
}